src/Publish/Microsoft.NET.Sdk.Publish.Tasks/WebConfigTransform.cs

```diff
@@ -26,9 +26,16 @@ public static XDocument Transform(XDocument webConfig, string appName, bool conf
             var rootElement = webConfig.Root.Element("location") == null ? webConfig.Root : webConfig.Root.Element("location");
             var webServerSection = GetOrCreateChild(rootElement, "system.webServer");
 
-            TransformHandlers(GetOrCreateChild(webServerSection, HandlersElementName));
+            var handlerSection = GetOrCreateChild(webServerSection, HandlersElementName);
+            TransformHandlers(handlerSection, aspNetCoreModuleName);
+
+            string aspNetCoreModuleValue =
+                    (string)handlerSection.Elements("add")
+                   .Single(e => string.Equals((string)e.Attribute("name"), "aspnetcore", StringComparison.OrdinalIgnoreCase))
```

Comment:
This might do well to be changed to FirstOrDefault to account for the possibility of it not being there & not having to assert that there's exactly one match

Reply:
If there is more than one module defined with the same name, then it would fail at runtime. I agree that I am being aggressive by failing it at build time. Will change it to account for none or more matches.